interface/reports/patient_list_creation.php

```diff
@@ -168,58 +194,107 @@ function sortingCols(sort_by,sort_order)
                     $('#ins_co').show();
-                <?php } ?>
+                <?php }
+                if (!empty($_POST['srch_option']) && ($_POST['srch_option'] == "Encounters")) { ?>
+                    $('#enc_type').show();
+                <?php }
+                if (!empty($_POST['srch_option']) && ($_POST['srch_option'] == "Observations")) { ?>
+                    $('#obs_desc').show();
+                <?php }
+                ?>
 
                 $('.datetimepicker').datetimepicker({
                     <?php $datetimepicker_timepicker = true; ?>
                     <?php $datetimepicker_showseconds = true; ?>
                     <?php $datetimepicker_formatInput = true; ?>
-                    <?php require($GLOBALS['srcdir'] . '/js/xl/jquery-datetimepicker-2-5-4.js.php'); ?>
+                    <?php require $GLOBALS['srcdir'] . '/js/xl/jquery-datetimepicker-2-5-4.js.php'; ?>
                     <?php // can add any additional javascript settings to datetimepicker here; need to prepend first setting with a comma ?>
                 });
+
+                // Automatically add wildcards to applicable inputs if they are empty
+                $('.wildcard_field').on('focus', function() {
+                    var field_value = $(this).val();
+                    if (field_value === '') {
+                        $(this).val('%%');
+                        var field = $(this)[0];
+                        field.setSelectionRange(1, 1);
+                    }
+                });
             });
 
-            function printForm(){
-                 var win = top.printLogPrint ? top : opener.top;
-                 win.printLogPrint(window);
+            function printForm() {
+                var win = top.printLogPrint ? top : opener.top;
+                win.printLogPrint(window);
             }
 
             function srch_option_change(elem) {
                 $('#sortby').val('');
                 $('#sortorder').val('');
 
-                if(elem.value == 'Communication') {
-                    $('#communication').val('');
+                // Reset and show/hide inputs related to specific search options
+                if (elem.value == 'Allergies' || elem.value == 'Problems' || elem.value == 'Medications' || elem.value == 'Procedures' || elem.value == 'Lab Results') {
+                    $('#pr_diag').show();
+                } else {
+                    $('#procedure_diagnosis').val('');
+                    $('#pr_diag').hide();
+                }
+                if (elem.value == 'Prescriptions') {
+                    $('#rx_drug').show();
+                } else {
+                    $('#prescription_drug').val('');
+                    $('#rx_drug').hide();
+                }
+                if (elem.value == 'Communication') {
+                    $('#communication').val('<?php echo xlt('All'); ?>');
```

Comment:
in the javascript context, should translate with xlj() function and when do this, then do not place the single quotes around it. For example:
```
$('#communication').val(<?php echo xlj('All'); ?>);
```